framework/tests/src/ibc.rs

```diff
@@ -0,0 +1,215 @@
+                                        description: None,
+                                        link: None,
+                                        origin: None,
+                                    },
+                                    vec![],
+                                )?
+                                .into()],
+                            })?,
+                        },
+                    },
+                    callback_info: None,
+                    retries: 2,
+                }],
+            })?,
+            PROXY.to_string(),
+        )?;
+
+        // The create remote account tx is passed ?
+        rt.block_on(
+            interchain.await_ibc_execution(OSMOSIS.to_owned(), create_account_remote_tx.txhash),
+        )?;
+
+        // We verify there is a new local account
+
+        Ok(())
+    }
+}
+// We need to rewrite this because cosmrs::Msg is not implemented for IBC types
+
+use abstract_interface::{Abstract, AccountDetails, ManagerExecFns, ManagerQueryFns};
+use anyhow::Result as AnyResult;
+use cosmwasm_std::{to_binary, Empty};
+use cw_orch::{
+    deploy::Deploy,
+    prelude::{Daemon, InterchainEnv, TxHandler},
+};
+use tokio::runtime::Runtime;
+
+use crate::OSMOSIS;
+
+pub const TEST_ACCOUNT_NAME: &str = "account-test";
+pub const TEST_ACCOUNT_DESCRIPTION: &str = "Description of the account";
+pub const TEST_ACCOUNT_LINK: &str = "https://google.com";
+
+pub fn set_env() {
+    std::env::set_var("STATE_FILE", "daemon_state.json"); // Set in code for tests
+    std::env::set_var("ARTIFACTS_DIR", "../artifacts"); // Set in code for tests
+}
+
+pub fn create_test_remote_account(
+    rt: &Runtime,
+    origin: &Daemon,
+    origin_name: &str,
+    destination: &str,
+    interchain: &InterchainEnv,
+) -> AnyResult<AccountId> {
+    let origin_abstract = Abstract::load_from(origin.clone())?;
+
+    // Create a local account for testing
+    let account_name = TEST_ACCOUNT_NAME.to_string();
+    let description = Some(TEST_ACCOUNT_DESCRIPTION.to_string());
+    let link = Some(TEST_ACCOUNT_LINK.to_string());
+    origin_abstract.account_factory.create_new_account(
+        AccountDetails {
+            name: account_name.clone(),
+            description: description.clone(),
+            link: link.clone(),
+        },
+        abstract_core::objects::gov_type::GovernanceDetails::Monarchy {
+            monarch: origin.sender().to_string(),
+        },
+    )?;
+
+    // We need to register the ibc client as a module of the manager (account specific)
+    origin_abstract
+        .account
+        .manager
+        .install_module(IBC_CLIENT, &Empty {}, None)?;
+
+    // Now we send a message to the client saying that we want to create an account on osmosis
+    let register_tx = origin_abstract.account.manager.exec_on_module(
+        to_binary(&abstract_core::proxy::ExecuteMsg::IbcAction {
+            msgs: vec![abstract_core::ibc_client::ExecuteMsg::Register {
```

Comment:
Did we make a decision on removing this requirement? I think having the ability to do it is useful but if you perform an action on a remote account that doesn't exist yet it should be created for the user.

Reply:
Yes, working on that now